test/integration/extensibilityTests.ts

```diff
@@ -0,0 +1,158 @@
+import {assert, expect} from 'chai'
+import fetch from 'node-fetch'
+import {config} from '../../src/config'
+
+// Tests to focus on extra details the SPA may need to supply at runtime
+describe('ExtensibilityTests', () => {
+
+    const oauthAgentBaseUrl = `http://localhost:${config.port}${config.endpointsPrefix}`
+
+    it('Starting a login request with a runtime OpenID Connect parameter should include it in the request URL', async () => {
+
+        const options = {
+            extraParams: [
+                {
+                    key: 'prompt',
+                    value: 'login',
+                },
+            ],
+        }
+
+        const response = await fetch(
+            `${oauthAgentBaseUrl}/login/start`,
+            {
+                method: 'POST',
+                headers: {
+                    origin: config.trustedWebOrigins[0],
+                    'content-type': 'application/json',
+                },
+                body: JSON.stringify(options),
+            },
+        )
+
+        assert.equal(response.status, 200, 'Incorrect HTTP status')
+        const body = await response.json()
+        const authorizationRequestUrl = body.authorizationRequestUrl as string
+
+        expect(authorizationRequestUrl).contains(
+            `${options.extraParams[0].key}=${options.extraParams[0].value}`,
+            'The extra parameter was not added to the authorization request URL')
+    })
+
+    it('Starting a login request with multiple OpenID Connect parameters should include them in the request URL', async () => {
+
+        const options = {
+            extraParams: [
+                {
+                    key: 'max-age',
+                    value: '3600',
+                },
+                {
+                    key: 'ui_locales',
+                    value: 'fr',
+                },
+            ],
+        }
+
+        const response = await fetch(
+            `${oauthAgentBaseUrl}/login/start`,
+            {
+                method: 'POST',
+                headers: {
+                    origin: config.trustedWebOrigins[0],
+                    'content-type': 'application/json',
+                },
+                body: JSON.stringify(options),
+            },
+        )
+
+        assert.equal(response.status, 200, 'Incorrect HTTP status')
+        const body = await response.json()
+        const authorizationRequestUrl = body.authorizationRequestUrl as string
+        
+        options.extraParams.forEach((p: any) => {
+            expect(authorizationRequestUrl).contains(
+                `${p.key}=${p.value}`,
+                'The extra parameters were not added to the authorization request URL')
+        })
+    })
+
+    it('Starting a login request with the OpenID Connect acr_values parameter should include it in the request URL', async () => {
```

Comment:
I think the first two tests are enough. This one and the next one don't give any additional value, in my opinion. If we want to test whether more complex values are added to the URL I would add the acr_values and claims parameters to the second test.